zanata-war/src/main/java/org/zanata/action/ContactAdminAction.java

```diff
@@ -68,47 +68,75 @@
 
+    /**
+     * Send email to admin by registered user.
+     */
+    @Restrict("#{identity.loggedIn}")
+    public void send() {
+        String fromName = authenticatedAccount.getPerson().getName();
+        String fromLoginName = authenticatedAccount.getUsername();
+        String replyEmail = authenticatedAccount.getPerson().getEmail();
         subject = msgs.get("jsf.message.admin.inquiry.subject");
-        message = "";
-    }
 
-    public void send() {
         try {
-            EmailStrategy strategy = new ContactAdminEmailStrategy(
-                    fromLoginName, fromName, replyEmail, subject, message);
+            sendEmail(fromLoginName, fromName, replyEmail);
+        } catch (Exception e) {
+            sendEmailFailedNotification(e, fromName, fromLoginName, replyEmail,
+                    message);
+        }
+    }
 
-            facesMessages.addGlobal(emailServiceImpl.sendToAdmins(strategy,
-                    null));
+    /**
+     * Send email to admin by anonymous user.
+     */
+    public void sendAnonymous() {
+        String fromLoginName = ANONYMOUS_USER;
+        String fromName = getClientIp(); //client ip address
+
+        try {
+            sendEmail(fromLoginName, fromName, NO_REPLY_EMAIL);
         } catch (Exception e) {
-            StringBuilder sb =
-                    new StringBuilder()
-                            .append("Failed to send email with subject '")
-                            .append(subject)
-                            .append("' , message '").append(message)
-                            .append("'");
-            log.error(
-                    "Failed to send email: fromName '{}', fromLoginName '{}', replyEmail '{}', subject '{}', message '{}'",
-                    e, fromName, fromLoginName, replyEmail, subject, message);
-            facesMessages.addGlobal(sb.toString());
+            sendEmailFailedNotification(e, fromName, fromLoginName, NO_REPLY_EMAIL,
+                message);
         }
     }
+
+    private void sendEmail(String fromLoginName, String fromName,
+            String replyEmail) throws Exception {
+        EmailStrategy strategy = new ContactAdminEmailStrategy(
+            fromLoginName, fromName, replyEmail, subject, message);
+
+        facesMessages.addGlobal(emailServiceImpl.sendToAdmins(strategy,
+                null));
+    }
+
+    private void sendEmailFailedNotification(Exception e, String fromName,
+            String fromLoginName, String replyEmail, String message) {
+        StringBuilder sb = new StringBuilder()
+            .append("Failed to send email with subject '")
+            .append(subject)
+            .append("' , message '").append(message)
+            .append("'");
+        log.error(
+                "Failed to send email: fromName '{}', fromLoginName '{}', replyEmail '{}', subject '{}', message '{}'",
+                e, fromName, fromLoginName, replyEmail, subject, message);
```

Comment:
There are 5 placeholders in the log message and you have 6 arguments. It will only work if exception is put at last. See http://slf4j.org/faq.html#paramException